### PR TITLE
TypicalPersons: Make all methods and fields static #569

### LINKED ISSUE #569: TypicalPersons: Make all methods and fields static
TypicalPersons is a utility class that creates preset Persons for tests
to use. Some methods and fields are non-static.

This makes it troublesome for test classes to access these preset 
Persons as they have to create a TypicalPersons object prior to
accessing the Person objects.

Let's make all methods and fields in TypicalPersons to be static. Also,
let's update the callers to static import the methods and fields they
require from TypicalPersons.

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Fixes #569 

```
TypicalPersons is a utility class that creates preset Persons for tests
to use. Some methods and fields are non-static.

This makes it troublesome for test classes to access these preset
Persons as they have to create a TypicalPersons object prior to
accessing the Person objects.

Let's make all methods and fields in TypicalPersons to be static.
```

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/2] [TypicalPersons: Make all methods and fields static](https://github.com/se-edu/addressbook-level4/pull/574/commits/0872c3a630111c1adf28a608d79d3f868b6ac7dd)
* [v1 2/2] [TypicalPersons: Remove getTypicalPersons\(\)](https://github.com/se-edu/addressbook-level4/pull/574/commits/cacb0252c5b0b9af35f6bd684b8fc6a27ec0998a)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/801bae82eea7e2e6c514a5b80dae0d91538ea82c...cacb0252c5b0b9af35f6bd684b8fc6a27ec0998a))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/574/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v2

@Zhiyuan-Amos submitted v2 for review.

* [v2 1/3] [TypicalPersons: Make all methods and fields static](https://github.com/se-edu/addressbook-level4/pull/574/commits/f9c27cf45b32ebb5bb2dcfb2cc04516dcc0e5360)
* [v2 2/3] [TypicalPersons: Remove getTypicalPersons\(\)](https://github.com/se-edu/addressbook-level4/pull/574/commits/1b97737703af291734edfb0af2f1b307aad6eb39)
* [v2 3/3] [TypicalPersons: Add private constructor](https://github.com/se-edu/addressbook-level4/pull/574/commits/d975164a0286ecf736e205edf4dd59369ffa14e6)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/48d5acc77c3a4611e63b8bcc934d2d206ba0028e...d975164a0286ecf736e205edf4dd59369ffa14e6)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/ac1fb8da12c6eee42a09d604e908e5cca4eae503~1...ac1fb8da12c6eee42a09d604e908e5cca4eae503))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/574/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
Good day dear sir @yamgent,

Thanks very much for your kind and prompt review. I have made the amazing changes that you have requested, and I sincerely hope that you find my PR satisfactory and pleasing to your eyes.

Thank you once again dear sir.

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
     public static final ReadOnlyPerson[] TYPICAL_PERSONS =
             new ReadOnlyPerson[]{ALICE, BENSON, CARL, DANIEL, ELLE, FIONA, GEORGE};
 
+    private TypicalPersons() {} // prevents instantiation
```
Comment (yamgent) [inline src/test/java/seedu/address/testutil/TypicalPersons.java]: I am surprised that checkstyle didn't catch this? :P

```
// prevents instantiation
private TypicalPersons() {
}
```

**Pull request merged**